src/libstore/lock.cc

```diff
@@ -94,10 +94,13 @@ struct SimpleUserLock : UserLock
                 if (err == -1)
                     throw Error("failed to get list of supplementary groups for '%s'", pw->pw_name);
 
+                gids.resize(ngroups);
```

Comment:
Note this line appears above, maybe comment would be good?

Reply:
I am not sure what to comment here. "resize the veector to the number of groups that were returned" is basically what the code line says, or is this already uncovering some misunderstanding on my side what the code is supposed to do?

Reply:
@tfc I am wondering did is already change size since it was resized above? If so, can we delete the other resize? Just seems odd to need to resize twice.

Reply:
I see. Added a comment about this that would hopefully clarify that.